cpp/src/arrow/array/validate.cc

```diff
@@ -166,6 +166,80 @@ struct ValidateArrayImpl {
     return Status::OK();
   }
 
+  Status Visit(const Date64Type& type) {
+    RETURN_NOT_OK(ValidateFixedWidthBuffers());
+
+    using c_type = typename Date64Type::c_type;
+    if (full_validation) {
+      constexpr c_type kFullDay = 1000 * 60 * 60 * 24;
+      return VisitArrayDataInline<Date64Type>(
+          data,
+          [&](c_type date) {
+            if (date % kFullDay != 0) {
+              return Status::Invalid(type, date,
+                                     "ms does not represent a whole number of days");
```

Comment:
Is a whitespace missing between `type` and `date`?
Also, it should be _"does not represents"_ ('s' at the end).

Reply:
The status would read something like: "Date64Type 1234ms does not represent a whole number of days" which should be grammatically correct.

Reply:
I _think_ it will be "Date64Type1234ms does not represent a whole number of days" (note the missing space: whitespace is not implicitly inserted between elements).

Reply:
You are correct. I was referring to after the addition of whitespace.